proposals/nnnn-rpc-design-refactoring.md

```diff
@@ -0,0 +1,150 @@
+# RPC design refactoring
+
+* Proposal: [SDL-NNNN](nnnn-rpc-design-refactoring.md)
+* Author: [Alexander Kutsan](https://github.com/LuxoftAKutsan)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core]
+
+## Introduction
+
+This proposal is to rework the design of the SDL core application manager layer in order to:
+ - Reduce coupling between components
+ - Reduce responsibility of the ApplicationManager class
+ - Provide a plugin architecture for adding additional functionality and RPC groups easily.
+
+## Motivation
+
+The SDL core project is actively implementing new features and new functionality.  
+To keep the project maintainable, on track and to reduce risks of regression during implementation
+refactoring of application manager layer is necessary.  
+Currently, extending SDL functionality may require code duplication. 
```

Comment:
Can we delete this line as it looks like it was duplicate?

Reply:
@mrapitis done in 7bb2896639e1d74563efc7ff368674f9894b3f0c